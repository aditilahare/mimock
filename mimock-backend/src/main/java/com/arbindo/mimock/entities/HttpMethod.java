package com.arbindo.mimock.entities;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import javax.persistence.*;
import java.time.ZonedDateTime;

@NoArgsConstructor
@AllArgsConstructor
@Builder
@Getter
@Entity
@Table(name = "http_methods")
public class HttpMethod {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    @Schema(example = "1", description = "HTTP Method Id")
    private Long id;

    @Column(name = "method", nullable = false, length = 10)
    @Schema(example = "GET", description = "Name of the HTTP Method")
    private String method;

    @Column(name = "created_at", nullable = false)
    @CreationTimestamp
    @Schema(description = "Creation Timestamp")
    private ZonedDateTime createdAt;

    @Column(name = "updated_at")
    @UpdateTimestamp
    @Schema(description = "Update Timestamp")
    private ZonedDateTime updatedAt;

    @Column(name = "deleted_at")
    @Schema(description = "Delete Timestamp")
    private ZonedDateTime deletedAt;

    @Override
    public String toString() {
        return method;
    }
}