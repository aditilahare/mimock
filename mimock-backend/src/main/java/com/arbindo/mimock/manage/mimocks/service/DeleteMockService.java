package com.arbindo.mimock.manage.mimocks.service;

public interface DeleteMockService {

    boolean hardDeleteMockById(String mockId);

    boolean softDeleteMockById(String mockId);

    boolean deleteAllMocks();

    void flushDeletedMocks();
}
