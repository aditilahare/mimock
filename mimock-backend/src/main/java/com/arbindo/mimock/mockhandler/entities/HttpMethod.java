package com.arbindo.mimock.mockhandler.entities;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import javax.persistence.*;
import java.time.Instant;

@NoArgsConstructor
@AllArgsConstructor
@Builder
@Getter
@Entity
@Table(name = "http_methods")
public class HttpMethod {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Long id;

    @Column(name = "method", nullable = false, length = 10)
    private String method;

    @Column(name = "created_at", nullable = false)
    @CreationTimestamp
    private Instant createdAt;

    @Column(name = "updated_at")
    @UpdateTimestamp
    private Instant updatedAt;

    @Column(name = "deleted_at")
    private Instant deletedAt;
}
