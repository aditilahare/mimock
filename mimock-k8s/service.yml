apiVersion: v1
kind: Service
metadata:
  name: mimock-svc
spec:
  type: NodePort
  selector:
    app: mimock
  ports:
    - protocol: TCP
      port: 8080
      targetPort: 8080
      nodePort: 30001
