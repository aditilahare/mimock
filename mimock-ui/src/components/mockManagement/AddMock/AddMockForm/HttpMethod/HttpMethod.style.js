import tw from 'tailwind-styled-components';

export const FormLabel = tw.label`
    w-1/5
    font-sans
    font-bold
    subpixel-antialiased	
    text-lg
    text-gray-600
`;
