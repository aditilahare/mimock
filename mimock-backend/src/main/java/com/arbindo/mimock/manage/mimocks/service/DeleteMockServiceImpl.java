package com.arbindo.mimock.manage.mimocks.service;

import com.arbindo.mimock.common.services.EntityStatusService;
import com.arbindo.mimock.entities.EntityStatus;
import com.arbindo.mimock.entities.Mock;
import com.arbindo.mimock.repository.MocksRepository;
import com.arbindo.mimock.utils.ValidationUtil;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.extern.log4j.Log4j2;
import org.apache.logging.log4j.Level;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import javax.transaction.Transactional;
import java.time.ZonedDateTime;
import java.util.List;

@Service
@Log4j2
@Builder
@AllArgsConstructor
public class DeleteMockServiceImpl implements DeleteMockService {

    @Autowired
    private MocksRepository mocksRepository;

    @Autowired
    private EntityStatusService entityStatusService;

    @Autowired
    private MockManagementService mockManagementService;

    @Override
    public boolean hardDeleteMockById(String mockId) {
        if (ValidationUtil.isNotNullOrEmpty(mockId)) {
            try {
                Mock mock = mockManagementService.getMockById(mockId);
                if (mock != null) {
                    mocksRepository.delete(mock);
                    return true;
                }
            } catch (Exception e) {
                log.log(Level.DEBUG, e.getMessage());
                return false;
            }
        }
        log.log(Level.DEBUG, "Invalid Mock Id!");
        return false;
    }

    @Transactional
    @Override
    public boolean softDeleteMockById(String mockId) {
        if (ValidationUtil.isNotNullOrEmpty(mockId)) {
            try {
                Mock mock = mockManagementService.getMockById(mockId);
                if (mock != null) {
                    // Perform only soft delete i.e. Mark EntityStatus as DELETED
                    EntityStatus entityStatus = entityStatusService.getDeletedMockEntityStatus();
                    mock.setEntityStatus(entityStatus);
                    mock.setDeletedAt(ZonedDateTime.now());
                    mocksRepository.save(mock);
                    return true;
                }
            } catch (Exception e) {
                log.log(Level.DEBUG, e.getMessage());
                return false;
            }
        }
        log.log(Level.DEBUG, "Invalid Mock Id!");
        return false;
    }

    @Override
    public boolean deleteAllMocks() {
        try {
            mocksRepository.deleteAll();
            return true;
        } catch (Exception e) {
            log.log(Level.DEBUG, e.getMessage());
        }
        log.log(Level.DEBUG, "Unable to delete all mocks!");
        return false;
    }

    @Override
    public void flushDeletedMocks() {
        try {
            EntityStatus entityStatus = entityStatusService.getDeletedMockEntityStatus();
            ZonedDateTime thirtyDaysAgo = ZonedDateTime.now().minusDays(30);
            List<Mock> deletedMocks = mocksRepository.findAllByEntityStatusAndDeletedAt(entityStatus, thirtyDaysAgo);
            mocksRepository.deleteAll(deletedMocks);
            log.log(Level.INFO, "Flushed " + deletedMocks.size() + " mock(s)!");
            return;
        } catch (Exception e) {
            log.log(Level.DEBUG, e.getMessage());
        }
        log.log(Level.DEBUG, "Unable to flush deleted mocks!");
    }
}
