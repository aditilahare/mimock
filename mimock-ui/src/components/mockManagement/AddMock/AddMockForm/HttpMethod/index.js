import HttpMethod from './HttpMethod';

export default HttpMethod;
