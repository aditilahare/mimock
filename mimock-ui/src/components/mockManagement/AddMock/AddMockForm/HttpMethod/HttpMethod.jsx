import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import { useRecoilState } from 'recoil';
import newMockFieldsAtom from 'atoms/newMockFieldsAtom';
import { getHttpMethods } from 'services/staticRecords/getHttpMethods.service';
import { Select } from 'styles';
import { FormItem, FormLabel, HttpMethodSelect } from '../AddMockForm.style';

function HttpMethod({ selectedMethod }) {
	const [mockData, setMockData] = useRecoilState(newMockFieldsAtom);
	const [httpMethods, setHttpMethods] = useState([]);

	useEffect(() => {
		getHttpMethods().then((methods) => {
			setHttpMethods(methods);
		});
	}, []);

	return (
		<If condition={selectedMethod && httpMethods && httpMethods.length}>
			<FormItem data-testid='http-method-selector'>
				<FormLabel data-testid='http-methods-label'>HTTP Method</FormLabel>
				<HttpMethodSelect>
					<Select
						options={httpMethods}
						defaultValue={selectedMethod}
						dataTestId='http-methods'
						onChange={(event) => {
							setMockData({ ...mockData, httpMethod: event.target.value });
						}}
					/>
				</HttpMethodSelect>
			</FormItem>
		</If>
	);
}

HttpMethod.propTypes = {
	selectedMethod: PropTypes.string,
};

HttpMethod.defaultProps = {
	selectedMethod: 'GET',
};

export default HttpMethod;
