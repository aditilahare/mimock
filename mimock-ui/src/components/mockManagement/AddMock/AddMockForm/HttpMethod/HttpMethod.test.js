import React from 'react';
import { act, render, fireEvent } from '@testing-library/react';
import { useRecoilState } from 'recoil';
import { getHttpMethods } from 'services/staticRecords/getHttpMethods.service';
import HttpMethod from './HttpMethod';

let mockedRecoilFn = jest.fn();
jest.mock('recoil');
jest.mock('services/staticRecords/getHttpMethods.service');

describe('HttpMethod', () => {
	beforeEach(() => {
		getHttpMethods.mockResolvedValue(['GET', 'POST', 'DELETE']);

		useRecoilState.mockImplementation(() => {
			return [
				{
					name: 'Tester',
					httpMethod: '',
				},
				mockedRecoilFn,
			];
		});
	});

	it('should render HttpMethod with methods fetched from api', async () => {
		let tree;

		await act(async () => {
			tree = await render(<HttpMethod selectedMethod='GET' />);
		});

		const { getByTestId, container } = tree;

		expect(getByTestId('http-method-selector')).toBeInTheDocument();
		expect(getByTestId('http-methods-label')).toHaveTextContent('HTTP Method');
		expect(getByTestId('http-methods')).toBeInTheDocument();

		expect(getByTestId('http-methods').value).toBe('GET');
		expect(getByTestId('http-methods').options).toHaveLength(3);

		expect(getHttpMethods).toHaveBeenCalledTimes(1);

		expect(container).toMatchSnapshot();
	});

	it('should set HTTP method on change', async () => {
		let tree;

		await act(async () => {
			tree = await render(<HttpMethod />);
		});

		const { getByTestId, container } = tree;

		expect(getByTestId('http-method-selector')).toBeInTheDocument();
		expect(getByTestId('http-methods-label')).toHaveTextContent('HTTP Method');
		expect(getByTestId('http-methods')).toBeInTheDocument();

		await act(async () => {
			fireEvent.change(getByTestId('http-methods'), {
				target: { value: 'POST' },
			});
		});

		expect(getByTestId('http-methods').value).toBe('POST');

		expect(getHttpMethods).toHaveBeenCalledTimes(1);

		expect(container).toMatchSnapshot();
	});
});
