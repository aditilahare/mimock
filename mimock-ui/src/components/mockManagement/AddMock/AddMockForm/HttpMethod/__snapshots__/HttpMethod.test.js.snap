// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HttpMethod should render HttpMethod with methods fetched from api 1`] = `
<div>
  <div
    class="flex items-center align-middle my-8"
    data-testid="http-method-selector"
  >
    <label
      class="w-1/5 font-sans font-bold subpixel-antialiased text-lg text-gray-600"
      data-testid="http-methods-label"
    >
      HTTP Method
    </label>
    <div
      class="mx-4"
    >
      <select
        class="font-sans rounded-md shadow-md font-semibold p-2 border-2 border-gray-200 outline-none"
        data-testid="http-methods"
      >
        <option
          data-testid="options"
          selected=""
          value="GET"
        >
          GET
        </option>
        <option
          data-testid="options"
          value="POST"
        >
          POST
        </option>
        <option
          data-testid="options"
          value="DELETE"
        >
          DELETE
        </option>
      </select>
    </div>
  </div>
</div>
`;

exports[`HttpMethod should set HTTP method on change 1`] = `
<div>
  <div
    class="flex items-center align-middle my-8"
    data-testid="http-method-selector"
  >
    <label
      class="w-1/5 font-sans font-bold subpixel-antialiased text-lg text-gray-600"
      data-testid="http-methods-label"
    >
      HTTP Method
    </label>
    <div
      class="mx-4"
    >
      <select
        class="font-sans rounded-md shadow-md font-semibold p-2 border-2 border-gray-200 outline-none"
        data-testid="http-methods"
      >
        <option
          data-testid="options"
          selected=""
          value="GET"
        >
          GET
        </option>
        <option
          data-testid="options"
          value="POST"
        >
          POST
        </option>
        <option
          data-testid="options"
          value="DELETE"
        >
          DELETE
        </option>
      </select>
    </div>
  </div>
</div>
`;
